lib/smack/Prelude.cpp

```diff
@@ -0,0 +1,438 @@
+#include "smack/Prelude.h"
+#include "smack/Naming.h"
+#include "smack/SmackOptions.h"
+#include "smack/Regions.h"
+
+namespace smack {
+
+const std::vector<unsigned> INTEGER_SIZES = {1, 5, 6, 8, 16, 24, 32, 40, 48,
+  56, 64, 80, 88, 96, 128, 160, 256};
+
+const std::vector<unsigned> REF_CONSTANTS = {
+  0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14,
+  1024
+};
+
+const std::vector<std::string> BINARY_OPS {"add", "sub", "mul", "sdiv",
+  "smod", "srem", "udiv", "urem", "shl", "lshr", "ashr", "and", "or", "xor",
+  "nand"};
+
+const std::vector<std::string> COMP_OPS {"ule", "ult", "uge", "ugt", "sle",
+  "slt", "sge", "sgt"};
+
+const std::vector<std::string> EQ_OPS {"eq", "ne"};
+
+const std::vector<BinExpr::Binary> BINARY_OPS_BODY {BinExpr::Plus,
+  BinExpr::Minus, BinExpr::Times};
+
+const std::vector<BinExpr::Binary> COMP_OPS_BODY {BinExpr::Lte, BinExpr::Lt,
+  BinExpr::Gte, BinExpr::Gt, BinExpr::Lte, BinExpr::Lt, BinExpr::Gte,
+  BinExpr::Gt};
+
+const std::vector<BinExpr::Binary> EQ_OPS_BODY {BinExpr::Eq, BinExpr::Neq};
+
+FuncDecl* Prelude::uninterpretedUnaryOp(int width, std::string name) {
+  std::string type = "i" + std::to_string(width);
+  return Decl::function(SmackRep::indexedName(name, {type}), {{"i1", type}},
+    type, NULL, {});
+}
+
+FuncDecl* Prelude::uninterpretedBinaryOp(int width, std::string name) {
+  std::string type = "i" + std::to_string(width);
+  return Decl::function(SmackRep::indexedName(name, {type}),
+    {{"i1", type}, {"i2", type}}, type, NULL, {});
+}
+
+FuncDecl* Prelude::inlineBinaryOp(int width, std::string name,
+                                  const Expr* body) {
+  std::string type = "i" + std::to_string(width);
+  return Decl::function(SmackRep::indexedName(name, {type}),
+    {{"i1", type}, {"i2", type}}, type, body, {Attr::attr("inline")});
+}
+
+FuncDecl* Prelude::builtinBinaryOp(int width, std::string builtinName,
+                                   std::string name) {
+  std::string type = "i" + std::to_string(width);
+  return Decl::function(SmackRep::indexedName(name, {type}),
+    {{"i1", type}, {"i2", type}}, type, NULL,
+    {Attr::attr("builtin", builtinName)});
```

Comment:
`builtin` or `bvbuiltin` are like magic numbers, too. Make the creation of `builtin` attributes inside a function. Imagine you have to change these attributes, e.g., per changes of Boogie spec. Then you only need to change the names in these functions rather than replace every string in their occurrences.